src/TableOfContents/index.tsx

```diff
@@ -214,6 +214,8 @@ export function TableOfContents<T extends TableOfContentsItem = TableOfContentsI
             className={cn(`m-${padding}`, { 'mr-0': hasFilter })}
           >
             <Button
+              className="truncate"
+              style={{ maxWidth: '12rem' }}
```

Comment:
I'm not sure this second line here is applicable globally. This `12rem` is an arbitrary number that looks good in that single case in ninja, but may not look good for others. What about defining it as `flex-basis: 0; flex-grow: ?` on both components (this and the text filter).  You can maybe add a `minWidth` / `maxWidth` as well but I wouldn't make those the primary width attributes.